CMake + cross-compilation doesn't work for HDF5Examples.

**Describe the bug**

```
  -- HDF5 H5_LIBVER_DIR: 116
  -- The CXX compiler identification is unknown
  CMake Error at HDF5Examples/C/CMakeLists.txt:2 (PROJECT):
    No CMAKE_CXX_COMPILER could be found.
  
    Tell CMake where to find the compiler by setting either the environment
    variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
    to the compiler, or to the compiler name if it is in the PATH.
```

**Expected behavior**

`cmake -DHDF5_ENABLE_SZIP_SUPPORT:BOOL=OFF ..` command works.

**Platform (please complete the following information)**
 - HDF5 version: develop
 - OS and version: ubuntu latest 
 - Compiler and version: gcc for aarch64 
 - Build system (e.g. CMake, Autotools) and version: cmake
 - Any configure options you specified: https://github.com/hdfeos/hdf5/blob/develop/.github/workflows/arm64.yml
 - MPI library and version (parallel HDF5): n/a

**Additional context**

The arm64.yml worked fine until HDF5Examples was added.

https://github.com/hdfeos/hdf5/actions/runs/6793175557

--- a/HDF5Examples/C/CMakeLists.txt
+++ b/HDF5Examples/C/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required (VERSION 3.12)
-PROJECT (HDF5Examples_C C)
+project (HDF5Examples_C C)
 
 #-----------------------------------------------------------------------------
 # Build the C Examples

--- a/HDF5Examples/C/CMakeLists.txt
+++ b/HDF5Examples/C/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required (VERSION 3.12)
-project (HDF5Examples_C)
+PROJECT (HDF5Examples_C C)
 
 #-----------------------------------------------------------------------------
 # Build the C Examples

--- a/HDF5Examples/C/H5PAR/CMakeLists.txt
+++ b/HDF5Examples/C/H5PAR/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required (VERSION 3.12)
-PROJECT (H5PAR_C C)
+project (H5PAR_C C)
 
 #-----------------------------------------------------------------------------
 # Define Sources

--- a/HDF5Examples/C/H5PAR/CMakeLists.txt
+++ b/HDF5Examples/C/H5PAR/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required (VERSION 3.12)
-PROJECT (H5PAR_C)
+PROJECT (H5PAR_C C)
 
 #-----------------------------------------------------------------------------
 # Define Sources

--- a/HDF5Examples/C/Perf/CMakeLists.txt
+++ b/HDF5Examples/C/Perf/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required (VERSION 3.12)
-PROJECT (HDF5Examples_C_PERFORM C)
+project (HDF5Examples_C_PERFORM C)
 
 #-----------------------------------------------------------------------------
 # Define Sources

--- a/HDF5Examples/C/Perf/CMakeLists.txt
+++ b/HDF5Examples/C/Perf/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required (VERSION 3.12)
-PROJECT (HDF5Examples_C_PERFORM)
+PROJECT (HDF5Examples_C_PERFORM C)
 
 #-----------------------------------------------------------------------------
 # Define Sources

